### ISSUE TITLE
Add triples to extraction template that link model to instance

### ISSUE THREAD

**Issue by grechaw** [marklogic/entity-services#145]
A proposal for EA-4 is to add a triple to the envelope.

The purpose of this triple is to provide an explicit indexed mapping of one instance document to its type.

To construct an IRI for a given instance, concatenate

prune(baseUri) +   "/instance/" + unique value

where the pruned baseURI is simply baseUri with trailing '#' or '/' removed.

'unique value' is TBD.

This triple is enough to provide to SPARQL and the Optic API a method to get from the entity model to its instances.  With this triple in each envelope, developers and architects will be able to query for all instances of an entity type, and get back references to documents.    Since it uses rdf:type to tag the instance data, all RDF-based modeling and inference can be employed on these instance documents and classes.

**Comment by grechaw** [comment]
We will discuss this enhancement further in next WG

**Comment by grechaw** [comment]
From demo day I learned that TDE templates can now have both triples and rows templates in the same TDE.  Also, fn:base-uri() is available within TDE templates.
These two facts together mean that I can implement this feature in the TDE template, rather than in the envelope transform.  A much cleaner solution, that integrates entity instances with semantics very nicely.

**Comment by grechaw** [comment]
Taking another look at modifying extraction-template-generate to implement this enhancement.  It is possible that, with TDE, we can also tie an instance to its document uri

**Comment by grechaw** [comment]
I will provide examples of these triples as part of design phase this week.

**Comment by grechaw** [comment]
Made good progress on prototyping this change.  Should be able to update spec tomorrow with some examples.

**Comment by grechaw** [comment]
Took back -- should use random integer rather than a required field for generating subject iris.

**Comment by grechaw** [comment]
After the discussion today, it seems that if there is no primary key, we just should not attempt to emit the two triples.  There's no reliable way to establish unique identity that will last, so the case for creating triples is kind of weak.

Proposing thus that we only emit the two triples if there is a primary key designated for the given type.

**Comment by bsrikan** [comment]
Extraction template in description generates a graph Uri which earlier was of the format 
baseUri#ETTitle-Version if baseUri did not end with "/"

Now its baseUri/ETTitle-Version irrespective of whether baseUri ends with "/" or not.
Please fix this @grechaw as discussed.

**Comment by grechaw** [comment]
OK back to you for verification.

**Comment by bsrikan** [comment]
Keys updated and new model added. Updated keys after #171 was merged. Changes available in PR #175

**Comment by bsrikan** [comment]
Shipping this enhancement as the tests are running OK in regression. Have added a note to self in e2e task in bugtrack to write Sparql and Optic queries for this enhancement. 
Verified with sample query in races example and get below

```
instance	                                                                       docUri	                                                      type
<http://grechaw.github.io/entity-types/Race-0.0.1/Runner/Bobbi>	"/races/data/race-data/races/10k1.xml"^^xs:anyURI	<http://grechaw.github.io/entity-types/Race-0.0.1/Runner>

```
